proposals/tasks/Explainer.md

```diff
@@ -1,138 +1,218 @@
 
-Associated with tasks are a new type:
+A `fiber` is a resource that is used to enable a computation that has an extent in time and has identity. The latter allows applications to manage their own computations and support patterns such as async/await, green threads and yield-style generators. 
+
+Associated with fibers is a new reference type:
 
 ```
-taskref
+fiber r0 .. rk
 ```
-which denotes tasks.
+which denotes fibers that return a vector of values of types `r0` through `rk`.
+
+#### The state of a fiber
+
+A fiber is inherently stateful; as the computation proceeds the fiber's state
+reflects that evolution. The state of a fiber includes function call frames and
+local variables that are currently live within the computation as well as
+whether the fiber is currently being executed. In keeping with the main design
+principles of WebAssembly, the frames and local variables represented within a
+fiber are not accessible other than by normal operations.
+
+In addition to the frames and local variables, it is important to know whether a
+fiber is suspendable or not. In essence, only fibers that have been explicitly
+suspended may be resumed, and ony fibers that are either currently executing or
+have resumed computations that are executing may be suspended.
+
+In general, validating that a suspended fiber may be resumed is a constant time
+operation but validating that an executing fiber may be suspended involves
+examining the chain of fiber resumptions.
 
-Notice that our `taskref` type is not qualified: tasks are not expected to return in the normal way that functions do. We explore this topic again in [Frequently Asked Questions](#frequently-asked-questions).
+Since a fiber may be referenced after it has completed, such references are not
+usable and the referenced fibers are asserted to be moribund.
 
+#### Fibers, ancestors and children
 
-#### The state of a task
+It is convenient to identify some relationships between fibers: the _resuming
+parent_ of a fiber is the fiber that most recently resumed that fiber. A _resume
+ancestor_ is either the _resume parent_ of the fiber, or is a resume ancestor of
+the resume parent of the fiber.
 
-A task is inherently stateful; as the computation proceeds the task's state reflects that evolution. The state of a task includes function call frames and local variables that are currently live within the computation as well as whether the task is currently being executed.
+The root of the ancestor relation is the _root task_ and represents the initial
+entry point into WebAssembly execution.
 
-However, apart from when a significant event occurs within the computation, the state of a task is not externally visible.
+The root fiber does not have an explicit identity. This is because, in general,
+child fibers cannot manage their ancestors.
 
-In our descriptions of task states it is convenient to identify an enumerated symbol that denotes the execution state of the task:
+### The `event` concept
```

Comment:
Quick question: are duplicate tags allowed within event handlers? Or must they be unique (say, to their enclosing handle block)?

Reply:
This is a level of detail that I had not considered. However, my inclination would be to forbid it within the same handle block.
There is a design question about whether event blocks can be inherited if you nest handle blocks. The way it is written at the moment, it would allow this; but in fact I am not sure it's a good idea. 
If you do allow inheritance, then, if you have a nested handle block with an event block of the same tag as an event block in an outer handle block, the inner one should hide the outer event block.